lib/slather/project.rb

```diff
@@ -51,7 +60,60 @@ def coverage_files
         dedupe(coverage_files)
       end
     end
-    private :coverage_files
+    private :gcov_coverage_files
+
+    def profdata_coverage_files
+      files = profdata_llvm_cov_output.split("\n\n")
+
+      files.map do |source|
+        coverage_file = coverage_file_class.new(self, source)
+        !coverage_file.ignored? ? coverage_file : nil
+      end.compact
+    end
+    private :profdata_coverage_files
+
+    def profdata_coverage_dir
+      if self.scheme
+        Dir["#{build_directory}/**/CodeCoverage/#{self.scheme}"].first
+      else
+        Dir["#{build_directory}/**/#{self.products.first.name}"].first
+      end
+    end
+
+    def binary_file
+      xctest_bundle_file = Dir["#{profdata_coverage_dir}/**/*.xctest"].first
+      if xctest_bundle_file == nil
+        raise StandardError, "No product binary found in #{profdata_coverage_dir}"
+      end
+
+      # Find the matching .app, if any
+      xctest_bundle_file_directory = Pathname.new(xctest_bundle_file).dirname
+
+      app_bundle_file = Dir["#{xctest_bundle_file_directory}/*.app"].first
+      framework_bundle_file = Dir["#{xctest_bundle_file_directory}/*.framework"].first
+      if app_bundle_file != nil
+        app_bundle_file_name_noext = Pathname.new(app_bundle_file).basename.to_s.gsub(".app", "")
+        "#{app_bundle_file}/#{app_bundle_file_name_noext}"
+      elsif framework_bundle_file != nil
+          framework_bundle_file_name_noext = Pathname.new(framework_bundle_file).basename.to_s.gsub(".framework", "")
+          "#{framework_bundle_file}/#{framework_bundle_file_name_noext}"
+      else
+        xctest_bundle_file_name_noext = Pathname.new(xctest_bundle_file).basename.to_s.gsub(".xctest", "")
+        "#{xctest_bundle_file}/#{xctest_bundle_file_name_noext}"
+      end
+    end
+    private :binary_file
+
+    def profdata_llvm_cov_output
+      profdata_coverage_dir = self.profdata_coverage_dir
+      if profdata_coverage_dir == nil || (coverage_profdata = Dir["#{profdata_coverage_dir}/**/Coverage.profdata"].first) == nil
+        raise StandardError, "No Coverage.profdata files found. Please make sure the \"Code Coverage\" checkbox is enabled in your scheme's Test action or the build_directory property is set."
+      end
+      xcode_path = `xcode-select -p`.strip
+      llvm_cov_command = File.join(xcode_path, "Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-cov show -instr-profile #{coverage_profdata} #{binary_file}")
+      `#{llvm_cov_command}`
```

Comment:
P.S. I'm the guy who mentioned enabling **Gather code coverage** (see below). You can see how we're doing it for the [Wikipedia-iOS app](https://github.com/wikimedia/wikipedia-ios), specifically [this commit](https://github.com/wikimedia/wikipedia-ios/commit/101a9a63cfe0cc4e17bcd5c8a72ebc6e877adfee).

![image](https://cloud.githubusercontent.com/assets/444217/9996971/c26e556c-6057-11e5-9170-66c732aa62cb.png)